test/src/test/java/org/corfudb/runtime/checkpoint/CheckpointSmokeTest.java

```diff
@@ -456,6 +462,293 @@ public void checkpointWriterInterleavedTest() throws Exception {
+                .build()
+                .begin();
+        Token snapshot = TransactionalContext
+                .getCurrentContext()
+                .getSnapshotTimestamp();
+        try {
+            cpw.startCheckpoint(snapshot);
+            cpw.appendObjectState(m.entryStream());
+            cpw.finishCheckpoint();
+
+            // Instantiate new runtime & map.
+            setRuntime();
+            Map<String, String> m2 = instantiateStringMap(streamName);
+            for (i = 0; i < numSmrEntries; i++) {
+                assertThat(m2.get(keyPrefix + i)).describedAs("get " + i)
+                        .isEqualTo(m.get(keyPrefix + i));
+            }
+            // 1 CheckpointEntry for SMREntries, 1 CheckpointEntry for finishCheckpoint()
+            assertThat(cpw.getNumEntries()).isEqualTo(2);
+        } finally {
+            r.getObjectsView().TXEnd();
+        }
+
+        // batchThresholdPercentage * maxWriteSize < (numBytesPerCheckpointEntry + lastEntrySize) < maxWriteSize
+        final int lastEntrySizeUpperBound = getRuntime().getParameters().getMaxWriteSize() - numBytesPerCheckpointEntry;
+        final int lastEntrySizeLowerBound = (int) Math.ceil(
+                getRuntime().getParameters().getMaxWriteSize() * batchThresholdPercentage - numBytesPerCheckpointEntry);
+        System.out.println("lastEntrySizeUpperBound: " + lastEntrySizeUpperBound);
+        System.out.println("lastEntrySizeLowerBound: " + lastEntrySizeLowerBound);
+
+        int left = 1;
+        int right = 1 << 20;
+        String lastPayload = "";
+        while (left < right) {
+            int mid = (left + right) / 2;
+            lastPayload = getRandomStringOfSize(mid);
+            int currentSize = getSerializedSMREntrySize(keyPrefix + i, lastPayload,
+                    cpw.getKeyMutator(), cpw.getValueMutator(), cpw.getSerializer());
+            if (currentSize > lastEntrySizeUpperBound) {
+                right = mid;
+            } else if (currentSize < lastEntrySizeLowerBound) {
+                left = mid + 1;
+            } else {
+                m.put(keyPrefix + numSmrEntries, lastPayload);
+                System.out.println(String.format("The last entry is: %s, its size is: %s",
+                        (keyPrefix + numSmrEntries), currentSize));
+                break;
+            }
+        }
+        numSmrEntries++;
+
+        // Write all CP data.
+        r.getObjectsView().TXBuild()
+                .type(TransactionType.SNAPSHOT)
+                .build()
+                .begin();
+        snapshot = TransactionalContext
+                .getCurrentContext()
+                .getSnapshotTimestamp();
+        try {
+            cpw.startCheckpoint(snapshot);
+            cpw.appendObjectState(m.entryStream());
+            cpw.finishCheckpoint();
+
+            // Instantiate new runtime & map.
+            setRuntime();
+            Map<String, String> m2 = instantiateStringMap(streamName);
```

Comment:
This will return the original map reference, its not a new map.

Reply:
Done. Please see the reply above.

Reply:
Resolving this conversation.